File: BackEnd/gamedex/src/main/java/cat/copernic/gamedex/entity/Commentary.java
package cat.copernic.gamedex.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.DBRef;
import lombok.Data;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;

@Document(collection = "commentary")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Commentary {

    @Id
    private String idCommentary;

    private StateType state;

    private String description;

    private double rating;

    @DBRef
    private User user;

    @DBRef
    private Videogame videogame;

    }

File: FrontEnd/app/src/main/java/cat/copernic/grup4/gamedex/Core/Model/Commentary.kt
package cat.copernic.grup4.gamedexandroid.Core.Model

import java.time.LocalDate

data class Commentary(

    val idCommentary: String,
    val state: StateType,
    val description: String,
    val rating: Double,
    val publishedDate: LocalDate,
    val user: User,
    //val videogame: Videogame

)
enum class StateType{
    PLAYING,WANTTOPLAY,FINISHED,DROPPED
}